File: src/services/student.service.js
import axios from 'axios';

const STUDENT_BASE_REST_API_URL = 'http://localhost:8080/api/v1/student';

class StudentService {

    saveStudent(student){
        return axios.post(STUDENT_BASE_REST_API_URL+"/add",student);
    }

    getAllStudents(){
        return axios.get(STUDENT_BASE_REST_API_URL);
    }
    
    deleteStudent(studentId){
        return axios.delete(STUDENT_BASE_REST_API_URL+`/delete/${studentId}`);
    }

    getStudentById(id){
        return axios.get(STUDENT_BASE_REST_API_URL + "/"+ id);
    }

    updateStudent(id,student){
        return axios.put(STUDENT_BASE_REST_API_URL + "/"+ id, student);
    }
}

export default new StudentService();

File: student-backend/src/main/java/com/student/demo/services/StudentService.java
package com.student.demo.services;

import java.util.List;

import com.student.demo.dto.StudentDto;
import com.student.demo.model.Student;

public interface StudentService {
	
Student createStudent(Student student);
	
	List<Student> getAllStudents();
	
	boolean deleteStudent(Long id);
	
	Student getstudentById(Long id);

	Student updateStudent(Long id, Student student);



}
